Update to build cew with espeak-ng

danielbair opened the pull request:
Tested on windows and mac.

Set AENEAS_USE_ESPEAKNG=True to build cew with espeak-ng

StephenMcConnel committed:
Add package for python3-aeneas

Also fix spurious warning messages that occur for at least python 3.8.

readbeyond committed:
Merge pull request #245 from StephenMcConnel/CreatePython3DebianPackage-20200402

Add package for python3-aeneas

danielbair committed:
Update to build cew with espeak-ng

danielbair committed:
Merge branch 'devel' into espeak-ng

danielbair reviewed (commented):
The debian changes are not mine...

danielbair commented:
espeak-ng installer for windows can be downloaded from their repository.

espeak-ng for mac is available from MacPorts and HomeBrew (from my homebrew-tap).

readbeyond commented:
@danielbair is this PR still relevant / needed / desired ?

danielbair commented:
On linux the espeak-ng packages provide a full replacement for espeak. But on Windows and Mac the espeak-ng package doesn't provide full espeak replacement files. So this PR does help Mac and Windows use espeak-ng with cew. I am implementing this PR when building the AIO installer, but this would allow others to benefit from this for installing via other methods.

dhouck commented:
On Arch linux it looks like `espeak-ng` now doesnʼt even fully replace `espeak` from the dev perspective, just the command-line tools (maybe thereʼs *something* in the AUR that does but I didnʼt see it), so I had to build without CEW even on linux.

abathur commented on setup.py:
```diff
@@ -156,6 +221,7 @@ def prepare_cew_for_windows():
 IS_LINUX = (os.name == "posix") and (os.uname()[0] == "Linux")
 IS_OSX = (os.name == "posix") and (os.uname()[0] == "Darwin")
 IS_WINDOWS = (os.name == "nt")
+IS_64BITS = (sys.maxsize > 2**32)
```
I had to patch in an `import sys` to get this working.

abathur reviewed (commented):
I can't speak to this changeset broadly, but I can at least note that the changes in https://github.com/readbeyond/aeneas/pull/258/commits/b80b1cacfc50f01535a1dff65ac7d4162d8b8868 helped me build aeneas with cew support via Nix. 

(The espeak package in nixpkgs isn't buildable on macOS since it hasn't been patched to disable the async feature [as homebrew does](https://github.com/Homebrew/homebrew-core/blob/84b589a59cffb5f64363af5820090846c983321d/Formula/e/espeak.rb#L43-L45). I wasn't sure about the consequences of patching the feature out, so I decided to try the espeak-ng approach first...)

readbeyond commented:
@danielbair is this PR still relevant / needed / desired ?

danielbair commented:
This was necessary to build cew with espeak-ng. Unless you have made some similar changes, then yes.

pettarin commented:
@danielbair This branch has conflicts (probably due to some of the merges I did today). Can you please take care of updating this PR, please? Then I will merge it in into ``develop`` trusting your judgement.

FYI I decided that the next version will v1.8.0 --- with the fixes and changes accumulated so far, mostly so that aeneas installs on modern versions of its dependencies (chiefly, Python, NumPy, espeak-ng).

danielbair committed:
Merge branch 'devel' into espeak-ng

danielbair commented:
The only conflict was in the debian/changelog which has the latest version listed as 1.7.4 (but you are stating that the next will be 1.8.0 so that will probably need to be changed.)

danielbair committed:
Update setup.py

Added `import sys` per comment https://github.com/readbeyond/aeneas/pull/258/changes/b80b1cacfc50f01535a1dff65ac7d4162d8b8868#r1359734746

danielbair committed:
Update copyright to current year.

Updated to current year.

danielbair commented:
This can be merged now.

readbeyond commented:
@danielbair thanks!
